datafusion/tests/sql.rs

```diff
@@ -5669,3 +5669,67 @@ async fn test_intersect_distinct() -> Result<()> {
     assert_batches_eq!(expected, &actual);
     Ok(())
 }
+
+#[tokio::test]
+async fn except_with_null_not_equal() {
+    let sql = "SELECT * FROM (SELECT null AS id1, 1 AS id2) t1
+            EXCEPT SELECT * FROM (SELECT null AS id1, 2 AS id2) t2";
+
+    let expected: Vec<Vec<String>> = vec![vec!["NULL".to_string(), "1".to_string()]];
```

Comment:
Using `&str` is possible here too I think, maybe easier to remove the explicit type.